test/EtomicSwap.js

```diff
@@ -47,593 +47,663 @@ const invalidSecretHex = '0x' + invalidSecret.toString('hex');
-        assert.equal(tokenBalanceBeforeReceiverSpend.toNumber(), amountToSend);
+        let tokenBalanceBeforeReceiverSpend = await erc1155token.balanceOf(etomicSwap.target, tokenId);
+        expect(tokenBalanceBeforeReceiverSpend).to.equal(BigInt(amountToSend));
 
         // Attempt to spend with invalid secret - should fail
-        await this.swap.receiverSpendErc1155(id, amountToSend, invalidSecretHex, this.erc1155token.address, tokenId, accounts[0], { from: accounts[1] }).should.be.rejectedWith(EVMThrow);
+        await etomicSwap.connect(accounts[1]).receiverSpendErc1155(id, amountToSend, invalidSecretHex, erc1155token.target, tokenId, accounts[0].address).should.be.rejected;
 
         // Attempt to claim from non-receiver address even with valid secret - should fail
-        await this.swap.receiverSpendErc1155(id, amountToSend, secretHex, this.erc1155token.address, tokenId, accounts[0], { from: accounts[0] }).should.be.rejectedWith(EVMThrow);
+        await etomicSwap.connect(accounts[0]).receiverSpendErc1155(id, amountToSend, secretHex, erc1155token.target, tokenId, accounts[0].address).should.be.rejected;
 
         // Successful spend by receiver with valid secret
-        await this.swap.receiverSpendErc1155(id, amountToSend, secretHex, this.erc1155token.address, tokenId, accounts[0], { from: accounts[1] }).should.be.fulfilled;
+        await etomicSwap.connect(accounts[1]).receiverSpendErc1155(id, amountToSend, secretHex, erc1155token.target, tokenId, accounts[0].address).should.be.fulfilled;
 
         // Check the state of the payment
-        const payment = await this.swap.payments(id);
-        assert.equal(payment[2].valueOf(), RECEIVER_SPENT);
+        const payment = await etomicSwap.payments(id);
+        expect(payment.state).to.equal(BigInt(RECEIVER_SPENT));
 
         // Check the balance of the token - should be transferred to the receiver (accounts[1])
-        let tokenBalance = await this.erc1155token.balanceOf(accounts[1], tokenId);
-        assert.equal(tokenBalance.toNumber(), amountToSend);
+        let tokenBalance = await erc1155token.balanceOf(accounts[1].address, tokenId);
+        expect(tokenBalance).to.equal(BigInt(amountToSend));
 
         // Check that the swap contract no longer holds the tokens
-        tokenBalance = await this.erc1155token.balanceOf(this.swap.address, tokenId);
-        assert.equal(tokenBalance.toNumber(), 0);
+        tokenBalance = await erc1155token.balanceOf(etomicSwap.target, tokenId);
+        expect(tokenBalance).to.equal(BigInt(0));
 
         // Attempting to spend again - should fail
-        await this.swap.receiverSpendErc1155(id, amountToSend, secretHex, this.erc1155token.address, tokenId, accounts[0], { from: accounts[1] }).should.be.rejectedWith(EVMThrow);
+        await etomicSwap.connect(accounts[1]).receiverSpendErc1155(id, amountToSend, secretHex, erc1155token.target, tokenId, accounts[0].address).should.be.rejected;
```

Comment:
Is it possible to check that a more specific error happened rather than plain promise rejection?

Reply:
sure, added `rejectedWith`. But I left `be.rejected` in `should allow to send ERC721 payment` and `should allow to send ERC1155 payment` tests, there are custom errors like:

```
VM Exception while processing transaction: reverted with custom error 'ERC721InsufficientApproval("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266", 1)'

VM Exception while processing transaction: reverted with custom error 'ERC1155InsufficientBalance("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266", 0, 1, 1)'
```

So I decided that we can leave some error description in comments, like this
https://github.com/KomodoPlatform/etomic-swap/blob/1a5465aaf1eff53c07c6fdb71aecc1134ccca1e1/test/EtomicSwap.js#L172-L173

Reply:
Yep, we can use matching from `expect().to.be.rejectedWith("Text which matches with part of error message");`